File: src/services/media-service/link.mapper.ts
import { ILink } from './types/link.type';

export class LinkMapper {

    public static map(response: any): Array<ILink> {
        return response.map((link: any) => LinkMapper.mapSingle(link));
    }

    public static mapSingle(link: any): ILink {
        return {
            identifier: link.Identifier,
            containerName: link.containerName,
            name: link.Name,
            url: link.Url,
            host: link.Host,
            port: link.Port,
            isSecure: link.IsSecure,
            iconUrl: link.IconUrl,
            category: link.Category,
            sortOrder: link.SortOrder
        };
    }
}

File: api/home-box-landing/HomeBoxLanding.Api/Features/Links/LinkMapper.cs
using HomeBoxLanding.Api.Features.Links.Types;

namespace HomeBoxLanding.Api.Features.Links;

public class LinkMapper
{
    public static Link Map(LinkRecord record)
    {
        return new Link
        {
            Identifier = record.Identifier,
            Name = record.Name,
            IconUrl = record.IconUrl,
            IsSecure = record.IsSecure,
            Port = record.Port,
            Host = record.Host,
            Url = record.Url,
            Category = record.Category,
            SortOrder = record.SortOrder
        };
    }
}